packages/collaboration-extension/src/index.ts

```diff
@@ -0,0 +1,204 @@
+} from '@jupyterlab/application';
+import { PageConfig } from '@jupyterlab/coreutils';
+import { DOMUtils } from '@jupyterlab/apputils';
+import {
+  AwarenessMock,
+  CollaboratorsPanel,
+  IAwareness,
+  ICurrentUser,
+  IGlobalAwareness,
+  IUserMenu,
+  IUserPanel,
+  RendererUserMenu,
+  RTCPanel,
+  User,
+  UserInfoPanel,
+  UserMenu
+} from '@jupyterlab/collaboration';
+import { usersIcon } from '@jupyterlab/ui-components';
+import { AccordionPanel, Menu, MenuBar } from '@lumino/widgets';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+import { IStateDB, StateDB } from '@jupyterlab/statedb';
+
+/**
+ * Jupyter plugin providing the ICurrentUser.
+ */
+const userPlugin: JupyterFrontEndPlugin<ICurrentUser> = {
+  id: '@jupyterlab/collaboration-extension:user',
+  autoStart: true,
+  provides: ICurrentUser,
+  activate: (app: JupyterFrontEnd): ICurrentUser => {
+    return new User();
+  }
+};
+
+/**
+ * Jupyter plugin providing the IUserMenu.
+ */
+const userMenuPlugin: JupyterFrontEndPlugin<IUserMenu> = {
+  id: '@jupyterlab/collaboration-extension:userMenu',
+  autoStart: true,
+  requires: [ICurrentUser],
+  provides: IUserMenu,
+  activate: (app: JupyterFrontEnd, user: ICurrentUser): IUserMenu => {
+    const { commands } = app;
+    return new UserMenu({ commands, user });
+  }
+};
+
+/**
+ * Jupyter plugin adding the IUserMenu to the menu bar if collaborative flag enabled.
+ */
+const menuBarPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/collaboration-extension:userMenuBar',
+  autoStart: true,
+  requires: [ICurrentUser, IUserMenu],
+  activate: (
+    app: JupyterFrontEnd,
+    user: ICurrentUser,
+    menu: IUserMenu
+  ): void => {
+    const { shell } = app;
+
+    if (PageConfig.getOption('collaborative') !== 'true') {
+      return;
+    }
+
+    const menuBar = new MenuBar({
+      forceItemsPosition: {
+        forceX: false,
+        forceY: false
+      },
+      renderer: new RendererUserMenu(user)
+    });
+    menuBar.id = 'jp-UserMenu';
+    user.changed.connect(() => menuBar.update());
+    menuBar.addMenu(menu as Menu);
+    shell.add(menuBar, 'top', { rank: 1000 });
+  }
+};
+
+/**
+ * Jupyter plugin creating a global awareness for RTC.
+ */
+const rtcGlobalAwarenessPlugin: JupyterFrontEndPlugin<IAwareness> = {
+  id: '@jupyterlab/collaboration-extension:rtcGlobalAwareness',
+  autoStart: true,
+  requires: [ICurrentUser, IStateDB],
+  provides: IGlobalAwareness,
+  activate: (
+    app: JupyterFrontEnd,
+    currentUser: User,
+    state: StateDB
+  ): IAwareness => {
+    const ydoc = new Y.Doc();
+
+    if (PageConfig.getOption('collaborative') !== 'true') {
+      return new AwarenessMock(ydoc);
+    }
+
+    const awareness = new Awareness(ydoc);
+
+    const server = ServerConnection.makeSettings();
+    const url = URLExt.join(server.wsUrl, 'api/yjs');
+
+    new WebsocketProvider(url, 'JupyterLab:globalAwareness', ydoc, {
```

Comment:
Looks like the handler exposed by `jupyter-server-ydoc` looks for a path: 

https://github.com/jupyter-server/jupyter_server_ydoc/blob/deb54d2b036bf02911367e9c9b38435c1966df65/jupyter_server_ydoc/ydoc.py#L33

Reply:
cc @davidbrochart who might have some input on this.

I think the original idea was to reuse the same endpoint for the global awareness. Although in that case there is no file. Wondering whether passing a placeholder would be enough, or maybe use a different approach.

Reply:
FWIW, with this rtc_left_panel branch merged with current master (4273278a5e3c174ba7469ef40f3e5cae43339376), I'm seeing an error in the referenced line of `jupyter-server-ydoc` when `path=JupyterLab:globalAwareness`:

```
[E 2022-07-29 13:35:53.436 ServerApp] Uncaught exception GET /api/yjs/JupyterLab:globalAwareness (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/api/yjs/JupyterLab:globalAwareness', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/site-packages/tornado/websocket.py", line 944, in _accept_connection
        await open_result
      File "/Users/jimgoo/git/jupyter_server_ydoc/jupyter_server_ydoc/ydoc.py", line 119, in open
        self.room = self.websocket_server.get_room(path)
      File "/Users/jimgoo/git/jupyter_server_ydoc/jupyter_server_ydoc/ydoc.py", line 60, in get_room
        file_format, file_type, file_path = path.split(":", 2)
    ValueError: not enough values to unpack (expected 3, got 2)
```

Reply:
Sorry, I missed that one @jtpio.
I need to see with @martinRenou, but I think we can make a special case for the global awareness, where the endpoint would just be `/api/yjs/globalAwareness`. Or maybe we can fake the file format and type with `/api/yjs/foo:bar:globalAwareness`. We basically want a Y room just for the awareness.

Reply:
@davidbrochart I managed to fake something out via:
```typescript
new WebsocketProvider(url, 'foo:bar:globalAwareness', ydoc, {
      awareness: awareness
});
```

Inside of `jupyter_server_ydoc` (https://github.com/jupyter-server/jupyter_server_ydoc/blob/main/jupyter_server_ydoc/ydoc.py#L59), this gives

```
ystore_class: <class 'jupyter_server_ydoc.ydoc.JupyterSQLiteYStore'>
path:  foo:bar:globalAwareness 
file_format: foo, file_type: bar, file_path: globalAwareness
updates_file_path: .bar:globalAwareness.y
```

At first I was seeing a lot of messages like `404 GET /api/yjs/foo:bar:globalAwareness (::1): file or directory does not exist: 'globalAwareness'`, but I just faked that out by doing:
```
touch globalAwareness
```

That seems to work fine as far as getting this branch merged with master working. I only notice the occasional messages like these, which I don't think are related to the faking out of globalAwareness: 
```
[I 2022-07-29 15:56:59.957 ServerApp] Creating new notebook in
[I 2022-07-29 15:57:00.224 ServerApp] Kernel started: 1f9f65aa-5eca-4337-8583-937f494129ba
Exception in callback <TaskWakeupMethWrapper object at 0x119b19a90>(<Future finished result=None>)
handle: <Handle <TaskWakeupMethWrapper object at 0x119b19a90>(<Future finished result=None>)>
Traceback (most recent call last):
  File "/Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
RuntimeError: Cannot enter into task <Task pending name='Task-859' coro=<WebsocketServer.serve() running at /Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/site-packages/ypy_websocket/websocket_server.py:103> wait_for=<Future finished result=None>> while another task <Task pending name='Task-96793' coro=<PeriodicCallback._run() running at /Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/site-packages/tornado/ioloop.py:921> cb=[IOLoop.add_future.<locals>.<lambda>() at /Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/site-packages/tornado/ioloop.py:694]> is being executed.
Task was destroyed but it is pending!
task: <Task pending name='Task-859' coro=<WebsocketServer.serve() running at /Users/jimgoo/miniconda3/envs/jupyterlab/lib/python3.8/site-packages/ypy_websocket/websocket_server.py:103> wait_for=<Future finished result=None>>```

The RTC left panel is amazing and works as advertised @martinRenou!

Reply:
Thanks for checking @jimgoo.
What you are seeing is jupyter-server-ydoc trying to save Y updates into a YStore, but I don't think this makes sense for the awareness. I'll update it to support rooms that don't have a file associated with them (feel free to open a PR if you feel like it).

> I only notice the occasional messages like these, which I don't think are related to the faking out of globalAwareness

I think this is a Tornado issue, I've been seeing that lately too.

Reply:
I released jupyter-server-ydoc v0.1.10. It should work with this PR.

Reply:
Thanks for the help @davidbrochart, the new version of jupyter-server-ydoc is working great. 

@martinRenou I took what you did in this branch and extended it to sync most of StateDB whenever a collaborator is clicked (https://github.com/jimgoo/jupyterlab/tree/rtc-layout-sync). It's kind of in the spirit of VSCode's LiveShare "Follow" feature, which I found had bugs with notebook files. I still want to sync a collaborator's scroll position inside of each widget as soon as I figure out how.

https://user-images.githubusercontent.com/2735842/182041267-36ec1819-4c45-4754-a10b-e9a09fae0147.mov

Reply:
Sorry I missed the notifications here.

Thank you @davidbrochart for the fix!

@jimgoo that's great! Would you like to push a PR against my branch to include your commits here?